### ISSUE TITLE
File seek error when adding a JPG file

### ISSUE THREAD

**Issue by cremor** [UglyToad/PdfPig#549]
I got a few JPG files that cause a seek exception when they are added to a PDF.

The exception is:
```
IOException: An attempt was made to move the file pointer before the beginning of the file.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.SeekCore(Int64 offset, SeekOrigin origin)
   at System.IO.FileStream.Seek(Int64 offset, SeekOrigin origin)
   at UglyToad.PdfPig.Images.JpegHandler.GetInformation(Stream stream)
   at UglyToad.PdfPig.Writer.PdfPageBuilder.AddJpeg(Stream fileStream, PdfRectangle placementRectangle)
```

I've tested both PdfPig version 0.1.7 and version 0.1.8-alpha-20230117-88aad, both show that exception.

Here is a sample file: https://user-images.githubusercontent.com/116314/218995467-ccb746b0-dc28-4616-bcb1-4ad685f81876.jpg

**Comment by fnatzke** [comment]
@cremor, 

If you can download and compile lib found the following fix by replacing this file

src\UglyToad.PdfPig\Images\JpegHandler.cs

with attached

[JpegHandler.zip](https://github.com/UglyToad/PdfPig/files/10752589/JpegHandler.zip)

In affect the loop to find the next marker uses the segment length rather than brute force try to find the MarkerStart (0xFF) of the next segment.

The Jpeg spec and consensus has there are number of known markers that are NOT followed by a segment length:
    StartOfImage
    EndOfImage
    Restart0
    Restart1
    Restart2
    Restart3
    Restart4
    Restart5
    Restart6
    Restart7
    TemporaryPrivateUseInArithmeticCoding

so special handling to not read two bytes as no length. For all other markers read the length and skip over the segment.

Will raise a pull request (PR).

**Comment by cremor** [comment]
Fixed by #551 in 0.1.8-alpha-20230219-999f9, thanks!
